src/Core/Resources/Resources.cpp

```diff
@@ -4,58 +4,70 @@
+#endif
+#if CXX_FILESYSTEM_IS_EXPERIMENTAL
+#    include <experimental/filesystem>
+
+// We need the alias from std::experimental::filesystem to std::filesystem
+namespace std {
+namespace filesystem = experimental::filesystem;
+}
+
+#else
+#    include <filesystem>
+#endif
+
 namespace Ra {
 namespace Core {
 namespace Resources {
 
-std::string searchPath( std::string pattern, std::string offset, void* libSymbol ) {
-    std::string baseDir = cpplocate::locatePath( pattern, offset, libSymbol );
-    std::replace( baseDir.begin(), baseDir.end(), '\\', '/' );
-    int nup = 0;
-    while ( baseDir.rfind( "../", baseDir.size() - 1 - 3 * nup ) != std::string::npos )
-    {
-        ++nup;
-    }
-    baseDir = baseDir.substr( 0, baseDir.size() - 3 * nup - 1 );
-    while ( nup-- )
-    {
-        baseDir = baseDir.substr( 0, baseDir.find_last_of( '/' ) + 1 );
-    }
-    return baseDir;
+std::filesystem::path
+searchPath( const std::string& pattern, const std::string& system, void* libSymbol ) {
+    std::string basePath = cpplocate::locatePath( pattern, system, libSymbol );
+    return std::filesystem::path( basePath ).lexically_normal();
 }
 
-std::string getRadiumResourcesDir() {
-    const std::string baseDir = searchPath(
-        "Resources/Shaders", "../..", reinterpret_cast<void*>( getRadiumResourcesDir ) );
-    return baseDir + "/Resources/";
+// add a trailing "/" if path point to an existing directory.
+std::filesystem::path clean( const std::filesystem::path& path ) {
+    auto status = std::filesystem::status( path );
+    if ( status.type() == std::filesystem::file_type::not_found ) return "";
+    if ( status.type() == std::filesystem::file_type::directory ) return path / "";
+    return path;
 }
 
-std::string getRadiumPluginsDir() {
-    const std::string baseDir =
-        searchPath( "Plugins/lib", "../../..", reinterpret_cast<void*>( getRadiumPluginsDir ) );
-    return baseDir + "/Plugins/lib/";
-}
+std::optional<std::string> getRadiumResourcesPath() {
+    auto basePath =
+        searchPath( "Resources/Shaders", "", reinterpret_cast<void*>( getRadiumResourcesPath ) );
 
-std::string getBaseDir() {
-    return cpplocate::getModulePath();
-}
+    basePath = clean( basePath / "Resources" );
 
-std::string getBaseResourcesDir() {
-    auto baseDir = cpplocate::locatePath( "Resources", "", nullptr );
-    return baseDir + "Resources/";
+    // first option
+    if ( basePath.empty() ) return {};
+    return ( basePath ).string();
 }
 
-ResourcesLocator::ResourcesLocator( void* symbol, const std::string& offset ) : m_basePath {""} {
-    m_basePath = searchPath( "/", "", symbol ) + offset;
+std::optional<std::string> getRadiumPluginsPath() {
+    auto basePath =
+        searchPath( "Plugins/lib", "", reinterpret_cast<void*>( getRadiumPluginsPath ) );
+    basePath = clean( basePath / "Plugins" / "lib" );
+    // second option
+    return basePath.empty() ? std::nullopt : std::optional<std::string>( basePath.string() );
```

Comment:
and the second... any pref ?

Reply:
well, short answer: no
` error: operands to ‘?:’ have different types ‘const std::nullopt_t’ and ‘std::filesystem::__cxx11::path’
   56 |     return basePath.empty() ? std::nullopt :  basePath ;`

Reply:
(see https://en.cppreference.com/w/cpp/utility/optional examples)

Reply:
Well, sorry. so I propose 
```suggestion
    return basePath.empty() ? std::nullopt : std::optional<std::string>( basePath );
```